### ISSUE TITLE
Support es module import statement

### ISSUE THREAD

**Issue by ebidel** [webcomponents/webcomponentsjs#994]
Currently you cannot simply `import` the loader like this:

```js
import 'https://unpkg.com/@webcomponents/webcomponentsjs@2.0.0/webcomponents-loader.js';
```

The loader requires a `<script>` tag. For projects that don't require bundling, it would be nice to support this way to load the lib.

**Comment by TimvdLippe** [comment]
We have recently added the `root` feature (https://github.com/webcomponents/webcomponentsjs#inlining) that allows you to say where the loader should fetch from. I think this can be used for your usecase? Maybe an extra example in the README could clarify this more.

**Comment by ebidel** [comment]
An example might help, I saw that section and dismissed it because I'm not "inlining".

However, I wasn't able to get this working with a cdn:

```
window.WebComponents = window.WebComponents || {};
window.WebComponents.root = 'https://unpkg.com/@webcomponents/webcomponentsjs@2.0.0/';
import 'https://unpkg.com/@webcomponents/webcomponentsjs@2.0.0/webcomponents-loader.js';
```

It still complains about script === null in FF.

**Comment by TimvdLippe** [comment]
Okay, so I had a fiddle with this and there were 2 issues I found:

1. The version that is loaded by the `import`  statement is 2.0.0. However, we have since shipped some updates. By loading the latest version (2.1.1 at the time of writing) with https://unpkg.com/@webcomponents/webcomponentsjs/webcomponents-loader.js we at least get the proper logic injected.
2. The second issue I found was that the assignment to `window.WebComponent` is ignored in a script `type="module"`. To counter that, the following snippet appears to be working correctly: http://jsbin.com/relehucuci/edit?html,console,output

I am not sure what is going on with 2 and whether this is a Firefox bug or not. We would need to repro this better with multiple files, but I have to run now. Hopefully this clarifies it more.

**Comment by ebidel** [comment]
I suspect 2 has something to do with how the polyfills are being loaded it's a race with modules being deferred by default. e.g. http://jsbin.com/sidimaqiru/edit?html,console,output

**Comment by TimvdLippe** [comment]
Yes that is true, but I would expect an `import` statement to not start before the other lines are finished with executing: http://jsbin.com/gezulucaxa/edit?html,console,output

**Comment by jsilvermist** [comment]
Aren't imports supposed to be at the top of a file/script, with nothing being loaded/set before them within that context?

**Comment by TimvdLippe** [comment]
@jsilvermist It only needs to be at the top scope, e.g. it can't be within an `if`-statement. However, it seems to be working as specced, as the following Plunkr shows the same in Chrome: https://plnkr.co/edit/wxXaJzSNYfn78aj4Ftg5?p=preview

**Comment by justinfagnani** [comment]
WHat is the reason a `<script>` tag isn't good enough? These polyfills have to be included at the top level of a page, so top-level import or script tag seem relatively equivalent. The loader especially has to run with blocking semantics, so the defer timing of modules won't work correctly.

**Comment by justinfagnani** [comment]
@TimvdLippe 

> Yes that is true, but I would expect an import statement to not start before the other lines are finished with executing: http://jsbin.com/gezulucaxa/edit?html,console,output

`import` is not a statement, and is not "started" or "executed". It's a declaration, and all imported modules run before the importing module.

**Comment by benoitjchevalier** [comment]
> WHat is the reason a `<script>` tag isn't good enough? These polyfills have to be included at the top level of a page, so top-level import or script tag seem relatively equivalent. The loader especially has to run with blocking semantics, so the defer timing of modules won't work correctly.

Trying to polyfill Shadow Dom for angular elements. Seems like loading the polyfill through a script tag is "too early" and the web components will run in the wrong zone, breaking angular change detection. The official way would be loading it synchronously in angular "polyfill.js" through an import statement

**Comment by TimvdLippe** [comment]
@benoitjchevalier Are you using the `root` option? If so, could you create a separate script tag to set that option and after that have the normal bundle with the `import` statement?

**Comment by benoitjchevalier** [comment]
Missed that thing about `root`, the approach you've described seems to have done the job, thanks @TimvdLippe !
